tests/Acoustics.Test/AnalysisPrograms/Recognizers/AustBitternTests.cs

```diff
@@ -0,0 +1,77 @@
+    /// Species name = Australasian Bittern = Botaurus poiciloptilus.
+    /// Recognizer class = BotaurusPoiciloptilus.cs.
+    /// </summary>
+    [TestClass]
+    public class AustBitternTests : OutputDirectoryTest
+    {
+        /// <summary>
+        /// The canonical recording used for this recognizer is a 30 second recording made by Liz Znidersic at Medeas Cove, St Helens, 2016-12-17.
+        /// </summary>
+        private static readonly FileInfo TestAsset = PathHelper.ResolveAsset("Recordings", "medeas_cove_2-2_1831_471228_20161217_232352_30_0.wav");
+        private static readonly FileInfo ConfigFile = PathHelper.ResolveConfigFile("RecognizerConfigFiles", "Towsey.BotaurusPoiciloptilus.yml");
+        private static readonly BotaurusPoiciloptilus Recognizer = new BotaurusPoiciloptilus();
+
+        //NOTE: If testing recording at its original sample-rate, then use line below.
+        //private static readonly AudioRecording Recording = new AudioRecording(TestAsset);
+
+        // If needing to resample, then must call AudioRecording.GetAudioRecording(TestAsset, resampleRate, opDir, opFileName);
+        //  as in the TestRecognizer() method below.
+
+        [TestMethod]
+        public void TestRecognizer()
+        {
+            var config = Recognizer.ParseConfig(ConfigFile);
+            int resampleRate = config.ResampleRate.Value;
+            string opDir = this.TestOutputDirectory.FullName;
+            string opFileName = "tempFile";
+            var recording = AudioRecording.GetAudioRecording(TestAsset, resampleRate, opDir, opFileName);
+
+            var results = Recognizer.Recognize(
+                audioRecording: recording,
+                config: config,
+                segmentStartOffset: TimeSpan.Zero,
+                getSpectralIndexes: null,
+                outputDirectory: this.TestOutputDirectory,
+                imageWidth: null);
+
+            var events = results.NewEvents;
+            var scoreTrack = results.ScoreTrack;
+            var plots = results.Plots;
+            var sonogram = results.Sonogram;
+
+            this.SaveTestOutput(
+                outputDirectory => GenericRecognizer.SaveDebugSpectrogram(results, null, outputDirectory, Recognizer.SpeciesName));
+
+            Assert.AreEqual(1, events.Count);
+            Assert.IsNull(scoreTrack);
+            Assert.AreEqual(1, plots.Count);
+            Assert.AreEqual(938, sonogram.FrameCount);
+
+            Assert.IsInstanceOfType(events[0], typeof(CompositeEvent));
+
+            var onlyEvent = (CompositeEvent)events[0];
+
+            Assert.AreEqual(5.12, onlyEvent.EventStartSeconds, 0.05);
+            Assert.AreEqual(12.26, onlyEvent.EventEndSeconds, 0.05);
+            Assert.AreEqual(105, onlyEvent.LowFrequencyHertz);
+            Assert.AreEqual(180, onlyEvent.HighFrequencyHertz);
+            Assert.AreEqual(21.7, onlyEvent.Score, 0.05);
+            Assert.AreEqual(0.95, onlyEvent.ScoreNormalized, 0.05);
```

Comment:
Remove the delta/tolerance unless it is needed. Maybe it is, but I doubt a tolerance of 0.05 is applicable to all of those wildly variable properties. 

A tolerance is added to a test to make it less brittle, particulrarly when a true/calculated expected value can not be represented because of some inherent calculation error (i.e. floating point error). Adding a tolerance implies such a problem and thus has meaning, hence why we should only add a tolerance where needed.

Reply:
Done

Reply:
Done